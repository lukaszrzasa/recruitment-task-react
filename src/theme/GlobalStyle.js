import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body, html {
    margin: 0;
    padding: 0;
    overflow: hidden;
  }
  * {
    overflow-wrap: break-word;
  }
`;

export default GlobalStyle;