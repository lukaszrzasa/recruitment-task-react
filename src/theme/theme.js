const colors = {
  violet: '#b33ed6',
  black: '#222',
  default: '#222',
  gray: '#809396',
  gold: '#ffdf13',
  white: '#fff',
  red: '#be2d10',
  green: '#13ad25',
  transparent: 'transparent',
};

const sizes = {
  xxl:200,
  xl: 60,
  lg:40,
  md:22,
  sm:18,
  xs: 14,
};

const theme = {
  colors,
  sizes,
};
export default theme;