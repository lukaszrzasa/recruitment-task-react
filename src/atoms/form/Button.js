import styled, {css} from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme/theme';
import {getColor} from '../helpers/getColor';



const Button = styled.button`
  width: ${theme.sizes.lg}px;
  height: ${theme.sizes.lg}px;
  line-height: ${theme.sizes.lg}px;
  text-align: center;
  border-radius: 50%;
  display: block;
  background-color: ${({variant}) => getColor(variant)};
  border: none;
  outline: none;
  
  ${({isSmall}) => isSmall && css`
    width: ${theme.sizes.sm}px;
    height: ${theme.sizes.sm}px;
    line-height: ${theme.sizes.sm}px;
  `}
`;

Button.propTypes = {
  isSmall: PropTypes.bool,
  variant: PropTypes.oneOf(Object.keys(theme.colors)).isRequired,
};

Button.defaultProps = {
  variant: theme.colors.default,
};


export default Button;