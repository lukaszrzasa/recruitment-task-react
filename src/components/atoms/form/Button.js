import styled, {css} from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../../theme/theme';
import {getColor} from '../helpers/getColor';
import { darken } from 'polished';


const {sizes, colors} = theme;

const Button = styled.button`
  width: ${sizes.xl}px;
  height: ${sizes.xl}px;
  line-height: ${sizes.xl}px;
  font-size: ${sizes.lg*0.8}px;
  text-align: center;
  border-radius: ${sizes.xl/2}px;
  background-color: ${({variant}) => getColor(variant)};
  color: ${({color}) => getColor(color)} !important;/*because of link*/
  border: none;
  outline: none;
  transition:background-color .3s;
  cursor: pointer;
  padding: 0;
  display: block;
  &::-moz-focus-inner {
    border: 0;
  }
  
  ${({isSmall}) => isSmall && css`
    width: ${sizes.sm}px;
    height: ${sizes.sm}px;
    line-height: ${sizes.sm}px;
    font-size: ${sizes.xs}px;
    border-radius: ${sizes.sm/2}px;
  `}
  
  &:hover {
    background-color: ${({variant}) => darken(.2,getColor(variant))};
  }
`;

export const TextButton = styled(Button)`
  width: unset !important;
  display: inline-block;
  padding: 5px 10px;
  height: ${sizes.sm+10}px;
  border-radius: ${(sizes.sm+10)/2}px;
  line-height: ${sizes.sm}px;
  font-size: ${sizes.xs}px;
`;

Button.propTypes = {
  isSmall: PropTypes.bool,
  variant: PropTypes.oneOf(Object.keys(colors)),
  color: PropTypes.oneOf(Object.keys(colors)),
};

Button.defaultProps = {
  variant: colors.default,
  color: colors.white
};


export default Button;