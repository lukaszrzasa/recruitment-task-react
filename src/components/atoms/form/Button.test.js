import {render} from '@testing-library/react';
import React from 'react';
import Button from './Button';
import theme from '../../../theme/theme';

const { colors, sizes } = theme;

describe('Button atom', () => {

  it('should render properly', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with assigned color/bg', () => {
    const { container, rerender } = render(<Button isSmall variant="red" color="gray" />);
    expect( container.querySelector('button') ).toHaveStyle(`background-color: ${colors.red}; color: ${colors.gray}; height: ${sizes.sm}px`);
    rerender(<Button color="white" />);
    expect( container.querySelector('button') ).toHaveStyle(`background-color: ${colors.default}; color: ${colors.white}; height: ${sizes.xl}px`);
    rerender(<Button variant="green" color="default" />);
    expect( container.querySelector('button') ).toHaveStyle(`background-color: ${colors.green}; color: ${colors.default};`);
  });

});