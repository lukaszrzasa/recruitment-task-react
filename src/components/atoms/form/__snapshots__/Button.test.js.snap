// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button atom should render properly 1`] = `
<button
  class="sc-AxjAm jZuqhl"
  color="white"
/>
`;
